### PR TITLE
User story 13

### LINKED ISSUE #13: User Story 13
Image

As a merchant
When I visit my items index page
Then next to each of the 5 most popular items I see the date with the most sales for each item.
And I see a label “Top selling date for <item name> was <date with most sales>"

Note: use the invoice date. If there are multiple days with equal number of sales, return the most recent day.

### PR DESCRIPTION
**Pull request by MatthewTLim**
# Describe the changes below:

# Relevant user story:

# Specific feedback requests? Include file name and line:

# Before submitting, check the following:
- [X] Entire test suite is passing
- [99.47%] SimpleCov test percentage

# Note any co-authors / anything else you'd like to add:

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -24,5 +24,19 @@ def ready_to_ship
...
   end
+
+  def item_revenue
+    success_ids = invoices.joins(:transactions).where(transactions: {result: "success"}).pluck(:id)
+    Item.select('name, SUM(invoice_items.unit_price * invoice_items.quantity) AS total_revenue').joins(invoices: :invoice_items).where(invoices: {id: success_ids}).group(:name).order("total_revenue desc")
+  end
+
+  def top_selling_dates_for_items
+    items
+      .select("items.*, COUNT(*) as sales_count, MAX(invoices.created_at) as top_selling_date")
```
Comment (dani-wilson) [inline app/models/merchant.rb]: Nice job on this

File: spec/features/merchant_spec.rb [hunk 2]
```diff
@@ -263,10 +267,10 @@
             @transaction_17 = @invoice_9.transactions.create!(credit_card_number: 4504301557459341, credit_card_expiration_date: "04/27", result: "success")
 
             visit merchant_items_path(@merchant_1)
-           
+            save_and_open_page
```
Comment (dani-wilson) [inline spec/features/merchant_spec.rb]: you left this here just FYI

**Review by westonio — APPROVED** [review]
Nice job!

**Review by dani-wilson — APPROVED** [review]
Good job good job!!

**Pull request closed without merging**